src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs

```diff
@@ -173,167 +171,134 @@ public async Task<CompletionResponse> Handle(CompletionRequest request)
-                                if (completion.DisplayText == Configuration.OmniSharpMiscProjectName) continue;
-                                CompletionChange change = await completionService.GetChangeAsync(document, completion);
-                                Debug.Assert(typedSpan == change.TextChange.Span);
-                                insertText = change.TextChange.NewText!;
-                            }
-                            break;
-
-                        case CompletionItemExtensions.XmlDocCommentCompletionProvider:
-                            {
-                                // The doc comment completion might compensate for the < before
-                                // the current word, if one exists. For these cases, if the token
-                                // before the current location is a < and the text it's replacing starts
-                                // with a <, erase the < from the given insertion text.
-                                var change = await completionService.GetChangeAsync(document, completion);
-
-                                bool trimFront = change.TextChange.NewText![0] == '<'
-                                                 && sourceText[change.TextChange.Span.Start] == '<';
-
-                                Debug.Assert(!trimFront || change.TextChange.Span.Start + 1 == typedSpan.Start);
-
-                                (insertText, insertTextFormat) = getAdjustedInsertTextWithPosition(change, position, newOffset: trimFront ? 1 : 0);
-                            }
-                            break;
-
-                        case CompletionItemExtensions.OverrideCompletionProvider:
-                        case CompletionItemExtensions.PartialMethodCompletionProvider:
+                    case CompletionItemExtensions.TypeImportCompletionProvider or CompletionItemExtensions.ExtensionMethodImportCompletionProvider:
+                        changeSpan = typedSpan;
+                        insertText = completion.DisplayText;
+                        seenUnimportedCompletions = true;
+                        sortText = '1' + completion.SortText;
+                        filterText = null;
+                        break;
+
+                    case CompletionItemExtensions.InternalsVisibleToCompletionProvider:
+                        // if the completion is for the hidden Misc files project, skip it
+                        if (completion.DisplayText == Configuration.OmniSharpMiscProjectName) continue;
+                        goto default;
+
+                    default:
+                        {
+                            // Except for import completion, we just resolve the change up front in the sync version. It's only expensive
+                            // for override completion, but there's not a heck of a lot we can do about that for the sync scenario
+                            var change = await completionService.GetChangeAsync(document, completion);
```

Comment:
does it now make sense to update the client to not call resolve on the server for each item except in specific cases?

Reply:
No, because resolve will still fill in documentation for all items.

Reply:
Any benefit (or problems!) with using `Task.WhenAll` to group all the async operations.  Not that we were doing this before anyway.

Reply:
I'd need to get a real benchmark setup to determine this. A quick test didn't show any _obvious_ impact either way.

Reply:
I've created a benchmark project using benchmarkdotnet and included in this PR. Some statistics from running it with and without `Task.WhenAll`:


|                    Method |     Mean |     Error |    StdDev |
|-------------------------- |---------:|----------:|----------:|
| ImportCompletionListAsyncOriginal | 3.715 ms | 0.0203 ms | 0.0190 ms |
| ImportCompletionListAsyncWhenAll | 4.232 ms | 0.0200 ms | 0.0178 ms |

|                  Method | NumOverrides |      Mean |    Error |   StdDev |    Median |
|------------------------ |------------- |----------:|---------:|---------:|----------:|
| OverrideCompletionAsyncOriginal |           10 |  11.54 ms | 0.051 ms | 0.045 ms |  11.53 ms |
| OverrideCompletionAsyncOriginal |          100 |  34.25 ms | 0.681 ms | 1.852 ms |  35.29 ms |
| OverrideCompletionAsyncOriginal |          250 |  65.92 ms | 1.312 ms | 3.194 ms |  67.20 ms |
| OverrideCompletionAsyncOriginal |          500 | 111.92 ms | 0.566 ms | 0.529 ms | 111.89 ms |
| OverrideCompletionAsyncWhenAll |           10 |  6.769 ms | 0.1011 ms | 0.0945 ms |
| OverrideCompletionAsyncWhenAll |          100 | 16.147 ms | 0.3139 ms | 0.2936 ms |
| OverrideCompletionAsyncWhenAll |          250 | 28.739 ms | 0.4544 ms | 0.3547 ms |
| OverrideCompletionAsyncWhenAll |          500 | 73.618 ms | 1.4516 ms | 1.7827 ms |
```